lib/fika/compiler/parser/function_def.ex

```diff
@@ -63,12 +79,49 @@ defmodule Fika.Compiler.Parser.FunctionDef do
     |> wrap(exps)
     |> concat(require_space)
     |> ignore(string("end"))
-    |> label("function definition")
-    |> Helper.to_ast(:function_def)
+    |> label("public function definition")
+    |> Helper.to_ast(:public_function_def)
 
-  function_defs =
+  ext_atom =
+    ignore(string("\""))
+    |> repeat(choice([string("\\\""), utf8_char(not: ?")]))
+    |> ignore(string("\""))
+    |> Helper.to_ast(:ext_atom)
+
+  ext_function_def =
     allow_space
-    |> concat(function_def)
+    |> ignore(string("ext"))
+    |> concat(require_space)
+    |> concat(identifier)
+    |> concat(arg_parens)
+    |> concat(return_type)
+    |> concat(allow_space)
+    |> ignore(string("="))
+    |> concat(allow_space)
+    |> ignore(string("{"))
+    |> concat(allow_space)
+    |> concat(ext_atom)
+    |> concat(allow_space)
+    |> ignore(string(","))
+    |> concat(allow_space)
+    |> concat(ext_atom)
+    |> concat(allow_space)
+    |> ignore(string(","))
+    |> concat(allow_space)
+    |> concat(arg_list)
+    |> concat(allow_space)
+    |> ignore(string("}"))
```

Comment:
I'd extract this as an MFA combinator or something like this, just to make the `ext_function_def` combinator more declarative